lib/internal/process.js

```diff
@@ -117,6 +118,35 @@ function setupMemoryUsage() {
   };
 }
 
+function setupResourceUsage() {
```

Comment:
It doesn't accept any parameters because it doesn't need any, it's just called once during startup.

Regarding array creation, I'm assuming you mean the `Float64Array()`? I copied how it was being done in other functions that use a similar pattern. I don't think there is any real benefit to moving it to C++. If anything it would be more complicated because of the additional C++ code that would be required (e.g. adding to `Persistent` list in src/env.h, etc.). Also, I personally would rather work more in JS than C++ ;-)